### ISSUE TITLE
aws_network_acl_rule icmp for IPv6 not working, icmpTypeCode missing

### ISSUE THREAD

**Issue by bbung** [terraform-providers/terraform-provider-aws#4737]
I try to add a ALL IPv6 ICMP Rule with icmp_type and icmp_code set to -1 as stated in the documentation an get the below error. I also can see the the metioned field is empty in the request using cloud trail.

```hcl
resource "aws_network_acl_rule" "private-dev-acl-ingress-icmpv6" {
  network_acl_id  = "${aws_network_acl.private-dev-acl.id}"
  rule_number     = 151
  egress          = false
  protocol        = 58                                      # icmp v6
  icmp_type       = -1
  icmp_code       = -1
  rule_action     = "allow"
  ipv6_cidr_block = "::/0"
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
+ provider.aws v1.21.0

### Affected Resource(s)

* aws_network_acl_rule

### Terraform Configuration Files

I am using a s3 bucket state and a special profile. Other rules or task are working fine with this configuration.
```hcl
provider "aws" {
  allowed_account_ids = [
    "########", 
  ]

  profile = "dev"

  region  = "${var.aws_region}"
  version = "~> 1.19"
}
```

### Expected Behavior

ICMP v6 Rule with 

151 | All ICMP - IPv6 | IPv6-ICMP (58) | ALL | ::/0 | ALLOW

### Actual Behavior

```
aws_network_acl_rule.private-dev-acl-ingress-icmpv6: Creating...
  egress:          "" => "false"
  icmp_code:       "" => "-1"
  icmp_type:       "" => "-1"
  ipv6_cidr_block: "" => "::/0"
  network_acl_id:  "" => "acl-XXXXX"
  protocol:        "" => "58"
  rule_action:     "" => "allow"
  rule_number:     "" => "151"

Error: Error applying plan:

1 error(s) occurred:

* aws_network_acl_rule.private-dev-acl-ingress-icmpv6: 1 error(s) occurred:

* aws_network_acl_rule.private-dev-acl-ingress-icmpv6: Error Creating Network Acl Rule: MissingParameter: The request must contain the parameter icmpTypeCode.type
        status code: 400, request id: ae04ee8a-943b-44b6-80a5-25720a4ad77e
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by bflad** [comment]
Pull request submitted: #6263

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.42.0 of the AWS provider, likely today or tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 1.42.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1420-october-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
